Git/syntax_test_gitconfig.gitconfig

```diff
@@ -0,0 +1,186 @@
+#   ^^^^^^^^^^^^^ variable.other.readwrite
+#                ^ -variable.other.readwrite
+#                 ^ keyword.operator.assignment
+#                  ^ -keyword.operator.assignment
+#                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.value
+#                   ^^^^^^^^^^^^ string.unquoted.value
+#                                ^^^^^^^^^ string.unquoted.value
+#                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ string.quoted.single
+#                                         ^ punctuation.definition.string.begin
+#                                                                                           ^ punctuation.definition.string.end
+    fatch = fetch --all --prune --tags
+# <- meta.key-value-pair
+#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^^^ variable.other.readwrite
+#        ^ -variable.other.readwrite
+#         ^ keyword.operator.assignment
+#          ^ -keyword.operator.assignment
+#           ^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.value
+#           ^^^^^ string.unquoted.value
+#                 ^^^^^ string.unquoted.value
+#                       ^^^^^^^ string.unquoted.value
+#                               ^^^^^^ string.unquoted.value
+    lgrep = log -1 -E -i --grep
+# <- meta.key-value-pair
+#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^^^ variable.other.readwrite
+#        ^ -variable.other.readwrite
+#         ^ keyword.operator.assignment
+#          ^ -keyword.operator.assignment
+#           ^^^^^^^^^^^^^^^^^^^ meta.value
+#           ^^^ string.unquoted.value
+#               ^^ string.unquoted.value
+#                  ^^ string.unquoted.value
+#                     ^^ string.unquoted.value
+#                        ^^^^^^ string.unquoted.value
+
+    pushf = push --force-with-lease
+# <- meta.key-value-pair
+#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^^^ variable.other.readwrite
+#        ^ -variable.other.readwrite
+#         ^ keyword.operator.assignment
+#          ^ -keyword.operator.assignment
+#           ^^^^^^^^^^^^^^^^^^^^^^^ meta.value
+#           ^^^^ string.unquoted.value
+#                ^^^^^^^^^^^^^^^^^^ string.unquoted.value
+    sla = log --oneline --decorate --graph --all
+# <- meta.key-value-pair
+#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.other.readwrite
+#      ^ -variable.other.readwrite
+#       ^ keyword.operator.assignment
+#        ^ -keyword.operator.assignment
+#         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.value
+#         ^^^ string.unquoted.value
+#             ^^^^^^^^^
+#                       ^^^^^^^^^^ string.unquoted.value
+#                                  ^^^^^^^ string.unquoted.value
+#                                          ^^^^^ string.unquoted.value
+
+# ESCAPES TEST
+    foo = "\\ \" \b \n \t"
```

Comment:
Maybe also add invalid.illegal for any other escapes like @jrappen suggested?

Reply:
I could make an `invalid.illegal` rule like this:
```yaml
- match: \\[rfv]
  scope: invalid.illegal.bad-escape.gitconfig
```
Unfortunately, this would mismatch valid text like `C:\Users\rob`. If the user intends that
`\r` represents a single carriage return char, that's a mistake. It would be nice if we could catch it. However, it might also be intended as two chars: a `\` followed by an `r`. In that case, it would be valid text. I can't think of a way to distinguish the two cases. The current behavior might be the best we can do.

Reply:
This:
```
[x]
    a = C:\Users\rob
```
prints
```
\fatal: bad config line 14 in file /Users/raoulwols/.gitconfig
```
while this
```
[x]
    a = C:\\Users\\rob
```
prints:
```
C:\Users\rob
```
but this is on macOS, not sure how git treats this on Windows?